src/hint_processor/bigint.zig

```diff
@@ -0,0 +1,234 @@
+    // Initiate BigInt default element
+    var p: BigInt3 = .{};
+    p = try p.fromVarName("P", vm, idsData, apTracking);
+    var p_packed = try p.pack(allocator);
+
+    std.debug.print("p_packed: {}\n", .{p_packed});
+
+    const p_packed_i256 = try p_packed.to(i256);
+    std.debug.print("p_packed_i256: {}\n", .{p_packed_i256});
+
+    var x: BigInt5 = .{};
+
+    const x_bigint5 = try x.fromVarName("x", vm, idsData, apTracking);
+    var x_lower: BigInt3 = .{};
+
+    // take first three limbs of x_bigint5 and pack them into x_lower
+    for (x_bigint5.limbs, 0..) |limb, i| {
+        if (i < 3) {
+            x_lower.limbs[i] = limb;
+        }
+    }
+
+    const x_lower_packed = try x_lower.pack(allocator);
+    std.debug.print("x_lower_packed: {}\n", .{x_lower_packed});
+    const d3 = x_bigint5.limbs[3].toInteger();
+    const d4 = x_bigint5.limbs[4].toInteger();
+    const x_lower_packed_256 = try x_lower_packed.to(i256);
+    std.debug.print("xxx: {}\n", .{x_lower_packed_256});
+    const c = x_lower_packed_256 + @as(i512, @intCast(d3)) * @as(i512, @intCast(std.math.pow(u512, BASE, 3))) + @as(i512, @intCast(d4)) * @as(i512, @intCast(std.math.pow(u512, BASE, 4)));
+    std.debug.print("c: {}\n", .{c});
+    const x_packed = c;
+
+    var y: BigInt3 = .{};
+
+    y = try y.fromVarName("y", vm, idsData, apTracking);
+
+    var y_packed = try y.pack(allocator);
+
+    const y_packed_i256 = try y_packed.to(i256);
+    const res = try divMod(x_packed, y_packed_i256, p_packed_i256);
+
+    try exec_scopes.assignOrUpdateVariable("res", .{
+        .i512 = res,
+    });
+    try exec_scopes.assignOrUpdateVariable("value", .{
+        .i512 = res,
+    });
+    try exec_scopes.assignOrUpdateVariable("x", .{
+        .i512 = x_packed,
+    });
+    try exec_scopes.assignOrUpdateVariable("y", .{
+        .i512 = y_packed_i256,
+    });
+    try exec_scopes.assignOrUpdateVariable("p", .{
+        .i512 = p_packed_i256,
+    });
+}
+
+/// Implements hint:
+/// ```python
+/// k = safe_div(res * y - x, p)
+/// value = k if k > 0 else 0 - k
+/// ids.flag = 1 if k > 0 else 0
+/// ```
+pub fn bigIntSafeDivHint(allocator: std.mem.Allocator, vm: *CairoVM, exec_scopes: *ExecutionScopes, ids_data: std.StringHashMap(HintReference), apTracking: ApTracking) !void {
+    const res = (try exec_scopes.getFelt("res")).toInteger();
+    const x = (try exec_scopes.getFelt("x")).toInteger();
+    const y = (try exec_scopes.getFelt("y")).toInteger();
+    const p = (try exec_scopes.getFelt("p")).toInteger();
+
+    const k = try safeDivBigInt(@as(i256, @intCast(res * y - x)), @as(i256, @intCast(p)));
+
+    var result: struct { value: i256, flag: Felt252 } = undefined;
+
+    if (k >= 0) {
+        result = .{ .value = k, .flag = Felt252.one() };
+    } else {
+        result = .{ .value = -k, .flag = Felt252.zero() };
+    }
+
+    const resultValue = result.value;
+    const flag = result.flag;
+
+    try exec_scopes.assignOrUpdateVariable("k", k);
```

Comment:
Here `assignOrUpdateVariable` is waiting for a `HintType` as last argument and not a `i256`. So that you have to transform your `i256` into a `HintType` by doing:

```zig
    try exec_scopes.assignOrUpdateVariable("k", .{.felt = Felt252.fromInt(u256, k}});
```

Reply:
Thank you